sdk/go/common/resource/plugin/plugin.go

```diff
@@ -513,12 +513,31 @@ func execPlugin(ctx *Context, bin, prefix string, kind apitype.PluginKind,
 		return nil, err
 	}
 
-	kill := func() error {
-		var result *multierror.Error
-
+	kill := sync.OnceValue(func() error {
 		// On each platform, plugins are not loaded directly, instead a shell launches each plugin as a child process, so
 		// instead we need to kill all the children of the PID we have recorded, as well. Otherwise we will block waiting
 		// for the child processes to close.
+
+		// Try to shut down gracefully first, so the plugins have a chance to clean up.
+		// If that fails we'll just kill the process.
+		cmdutil.InterruptChildren(cmd.Process.Pid)
+
+		// Give the process 500ms to shut down, or kill it forcibly.
+		timer := time.NewTimer(500 * time.Millisecond)
```

Comment:
Is that enough time?

Reply:
I *think* so, mostly this should just be simply exiting the plugin, without doing all that much cleanup. Probably doesn't hurt to make it a little longer though, since if the process exits quickly, we detect that.  Maybe 5 seconds?

Reply:
Yeah 5 seconds sounds good. If we're too impatient and move on to kill the processes,  we kinda lose the benefit of doing this in the first place.